sui_core/src/authority/authority_store.rs

```diff
@@ -968,14 +958,38 @@ impl<const ALL_OBJ_VER: bool> SuiDataStore<ALL_OBJ_VER> {
             .map_err(SuiError::from)
     }
 
+    pub fn get_transaction(
+        &self,
+        transaction_digest: &TransactionDigest,
+    ) -> SuiResult<Option<TransactionEnvelope<S>>> {
+        let transaction = self.transactions.get(transaction_digest)?;
+        Ok(transaction)
+    }
+
     #[cfg(test)]
     /// Provide read access to the `schedule` table (useful for testing).
     pub fn get_schedule(&self, object_id: &ObjectID) -> SuiResult<Option<SequenceNumber>> {
         self.schedule.get(object_id).map_err(SuiError::from)
     }
 }
 
-impl<const ALL_OBJ_VER: bool> BackingPackageStore for SuiDataStore<ALL_OBJ_VER> {
+impl<const A: bool> SuiDataStore<A, AuthoritySignInfo> {
+    pub fn get_signed_transaction_info(
+        &self,
+        transaction_digest: &TransactionDigest,
+    ) -> Result<TransactionInfoResponse, SuiError> {
+        Ok(TransactionInfoResponse {
+            signed_transaction: self.transactions.get(transaction_digest)?,
+            certified_transaction: self.certificates.get(transaction_digest)?,
+            signed_effects: self.signed_effects.get(transaction_digest)?,
+        })
+    }
+}
+
+impl<const A: bool, S: AuthoritySignInfoTrait> BackingPackageStore for SuiDataStore<A, S>
+where
```

Comment:
If the bound on `S` is `: Eq + ...` (which implies a `PartialEq` bound), and `TransactionEnvelope` derives `PartialEq` and `Eq`, then you don't need a `PartialEq` bound on `TransactionEnvelope<S>`, it's satisfied by derivation.